.travis.yml

```diff
@@ -27,25 +26,41 @@ deploy:
     skip_cleanup: true
     on:
       tags: true
+  - provider: script
+    script: echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin && docker buildx build --push -t ghostunnel/ghostunnel:${TRAVIS_TAG} --platform linux/amd64,linux/arm64 --build-arg VERSION=${TRAVIS_TAG} -f Dockerfile-multiarch .
+    skip_cleanup: true
+    on:
+      tags: true
+  - provider: script
+    script: echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin && docker buildx build --push -t ghostunnel/ghostunnel:latest --platform linux/amd64,linux/arm64 --build-arg VERSION=${TRAVIS_BRANCH} -f Dockerfile-multiarch .
+    skip_cleanup: true
+    on:
+      branch: master
 
 before_install:
   - sudo apt-get update
-  - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
+  - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker.io
+  - mkdir -vp ~/.docker/cli-plugins/
+  - curl --silent -L "https://github.com/docker/buildx/releases/download/v0.4.1/buildx-v0.4.1.linux-amd64" > ~/.docker/cli-plugins/docker-buildx
+  - chmod a+x ~/.docker/cli-plugins/docker-buildx
+  - docker buildx version
+  - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+  - docker buildx create --use
   - go get github.com/mattn/goveralls
   - go get github.com/karalabe/xgo
 
 install:
   # Compile with CGO on/off for testing
   - export GO111MODULE=on
-  - CGO_ENABLED=0 go build -o ghostunnel-${TRAVIS_TAG}-linux-amd64-without-pkcs11 .
-  - CGO_ENABLED=1 go build -tags nopkcs11 -o ghostunnel-${TRAVIS_TAG}-linux-amd64-without-pkcs11-but-with-cgo .
-  - CGO_ENABLED=1 go build -o ghostunnel-${TRAVIS_TAG}-linux-amd64-with-pkcs11 .
+  - CGO_ENABLED=0 go build -o ghostunnel-${TRAVIS_BRANCH}-linux-amd64-without-pkcs11 .
+  - CGO_ENABLED=1 go build -tags nopkcs11 -o ghostunnel-${TRAVIS_BRANCH}-linux-amd64-without-pkcs11-but-with-cgo .
+  - CGO_ENABLED=1 go build -o ghostunnel-${TRAVIS_BRANCH}-linux-amd64-with-pkcs11 .
+  # Cross-compile dist binaries
+  - GO_VERSION=${TRAVIS_GO_VERSION} VERSION=${TRAVIS_BRANCH} make -f Makefile.dist dist
   # Build Docker container
-  - make docker-build
+  - docker build -t ghostunnel/ghostunnel:${TRAVIS_BRANCH} --build-arg VERSION=${TRAVIS_BRANCH} --build-arg TARGETPLATFORM=linux/amd64 -f Dockerfile-multiarch .
```

Comment:
Also need to add arm64 here?

Reply:
This is because docker buildx is only capable of pushing to a registry at the moment (see https://github.com/docker/buildx/issues/59) so I've done a standard docker build so we can test the container runs. Then I have to populate the TARGETPLATFORM which is usually provided by buildx with something. However as in the above two comments if we want to go back to the compiling-in-build and using Alpine approach we won't need to use TARGETPLATFORM at all and everything will be simpler.

Reply:
You should be able to use buildx with `--load` if you only use the `linux/amd64` `--platform` arg, making it more like the later command. Although, I do think that the issue in question is actually resolved :)

Reply:
With multiple architectures and the latest version of buildx I still get 
![image](https://user-images.githubusercontent.com/11150054/90565108-42cc7780-e19e-11ea-824d-f491e303c30d.png)

:(

But yes you're right just specifying amd64 does work but basically makes it the same as a normal `docker build`